crates/dbsp/src/operator/communication/exchange.rs

```diff
@@ -44,6 +47,14 @@ use tokio::{
 };
 use typedmap::TypedMapKey;
 
+/// Current time in microseconds.
+fn current_time_usecs() -> u64 {
+    SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .map(|d| d.as_micros() as u64)
```

Comment:
How about make it nanoseconds to make it more consistent with other metrics we already have in ns?

Reply:
Again, it all gets converted to Duration. Internally, micros can be represented as AtomicU64, but nanos require u128.

Reply:
Problem is there's no AtomicU128. `try_into()` is better though, I'll do that.

Reply:
I don't think you need AtomicU128, 18446744073709551615 nanoseconds are ~580 yrs, but I commend the approach to robustness